projects/ngx-location-viewer/src/lib/services/layer.service.ts

```diff
@@ -172,7 +172,21 @@ export class LayerService {
 
     return layer;
   }
-  private buildChildLayer(parentLayerId: number, layers: LayerInfo[], layerLegend: LayerLegend[], visible?: boolean): Layer[] {
+
+  private determineLayerVisibility(layerId: number, defaultVisibility: boolean, visible?: boolean | number[]): boolean {
+    if (visible != null) {
+      if (typeof(visible) === 'boolean') {
+        return visible;
+      } else {
+        return visible.includes(layerId);
+      }
+    } else 
+    {
+      return defaultVisibility;
+    }
```

Comment:
Ik zou dit iets eenvoudiger schrijven:
```suggestion
    if (visible != null) {
      if (typeof(visible) === 'boolean') {
        return visible;
      }
      return visible.includes(layerId);
    }
    return defaultVisibility;
```